### PR TITLE
Our builds stopped working because latest is no longer working

### PR DESCRIPTION
**Pull request by wenlock**
This PR switches us to using glide to build the plugin and pins on specific versions
of the libraries that we require for the plugin.

We'll also change the Dockerfile build process to use the current source for our
builds vs trying to git clone it from the remote repos.

Signed-off-by: Edward Raigosa <edward.raigosa@hpe.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,18 +2,20 @@ FROM golang:alpine
...
     rm /usr/local/terraform/terraform-provisioner-chef
 
 RUN mkdir -p /go/src/github.com/HewlettPackard/ && \
-    cd /go/src/github.com/HewlettPackard && \
-    git clone https://github.com/HewlettPackard/terraform-provider-oneview.git && \
-    cd terraform-provider-oneview && \
-    git config --global http.sslVerify false && \
-    go get && \
+    cd /go/src/github.com/HewlettPackard/terraform-provider-oneview && \
+    wget --no-check-certificate https://glide.sh/get -O ./glide.sh && \
```
Comment (miqui) [inline Dockerfile]: this is how i do it to simply pin glide version:
```
ENV GLIDE_VERSION 0.12.1
ENV GLIDE_DOWNLOAD_URL https://github.com/Masterminds/glide/releases/download/v${GLIDE_VERSION}/glide-v${GLIDE_VERSION}-linux-amd64.tar.gz

RUN curl -fsSL "$GLIDE_DOWNLOAD_URL" -o glide.tar.gz \
    && tar -xzf glide.tar.gz \
    && mv linux-amd64/glide /usr/bin/ \
    && rm -r linux-amd64 \
    && rm glide.tar.gz
```

but it is not important, just to consider...

**Comment by gauravbansal17** [comment]
LGTM

**Comment by patrickdappollonio** [comment]
You need to bump the Go version so `sort.Slice` will work in the `.travis.yaml` file.

Also, if needed, we should pin the versions of those dependencies here, rather than in the build flow, so that way you guys will never need to pin them every time you create the container.

Let me know if you need help!

**Comment by wenlock** [comment]
@patrickdappollonio but isn't in the container that the libraries need to exist?  If we get them from the .travis.yml then when i build it, it would still pull from latest.   Let me try to switch to patch version and see if that makes the `sort.Slice` issue disappear for us.

**Comment by patrickdappollonio** [comment]
Yeah, in the docker container there will be no problem if you're using the `golang` image.

My point was, you might not want to always, every time you build, have to pull your dependencies, not from internet, not from glide. With Go, you can keep them along with the code and build with those, so the glide part doesn't become an "extra step" in the build process and you can safely remove it even from the Dockerfile.

**Comment by wenlock** [comment]
Lets add vendor folder after this merge, sorry I merged the wrong one earlier, i added a revert and merged that to undo it.

**Pull request merged**